integration/shared/isolated/marketplace_command_test.go

```diff
@@ -0,0 +1,336 @@
+							})
+
+							AfterEach(func() {
+								broker.Destroy()
+								helpers.QuickDeleteOrg(org)
+							})
+
+							It("displays extended information about the service", func() {
+								description := "Shared fake Server, 5tb persistent disk, 40 max concurrent connections"
+								session := helpers.CF("marketplace", "-s", getServiceName(broker))
+								Eventually(session).Should(Say("Getting service plan information for service %s as %s\\.\\.\\.", getServiceName(broker), user))
+								Eventually(session).Should(Say("OK"))
+								Eventually(session).Should(Say("\n\n"))
+								Eventually(session).Should(Say("service plan\\s+description\\s+free or paid"))
+								Eventually(session).Should(Say("%s\\s+%s\\s+%s", getPlanName(broker), description, "free"))
+								Eventually(session).Should(Exit(0))
+							})
+						})
+
+						When("the specified service is accessible but not in the targeted space", func() {
+							var (
+								broker helpers.ServiceBroker
+								org    string
+								space  string
+							)
+
+							BeforeEach(func() {
+								org = helpers.NewOrgName()
+								space = helpers.NewSpaceName()
+								helpers.SetupCF(org, space)
+								helpers.TargetOrgAndSpace(org, space)
+
+								domain := helpers.DefaultSharedDomain()
+
+								broker = createBroker(domain, "SERVICE", "SERVICE-PLAN")
+								enableServiceAccessForOrg(broker, org)
+
+								helpers.TargetOrgAndSpace(ReadOnlyOrg, ReadOnlySpace)
+							})
+
+							AfterEach(func() {
+								helpers.TargetOrgAndSpace(org, space)
+								broker.Destroy()
+								helpers.QuickDeleteOrg(org)
+							})
+
+							It("displays an error that the service doesn't exist", func() {
+								session := helpers.CF("marketplace", "-s", getServiceName(broker))
+								Eventually(session).Should(Say("Getting service plan information for service %s as %s\\.\\.\\.", getServiceName(broker), user))
+								Eventually(session).Should(Say("FAILED"))
+								Eventually(session.Err).Should(Say("Service offering '%s' not found", getServiceName(broker)))
+								Eventually(session).Should(Exit(1))
+							})
+						})
+					})
+				})
+			})
+		})
+	})
+})
+
+func createBroker(domain, serviceName, planName string) helpers.ServiceBroker {
+	service := helpers.PrefixedRandomName(serviceName)
+	servicePlan := helpers.PrefixedRandomName(planName)
+	broker := helpers.NewServiceBroker(helpers.NewServiceBrokerName(), helpers.NewAssets().ServiceBroker, domain, service, servicePlan)
+	broker.Push()
+	broker.Configure(true)
+	broker.Create()
+
+	return broker
+}
+
+func enableServiceAccess(broker helpers.ServiceBroker) {
+	Eventually(helpers.CF("enable-service-access", getServiceName(broker))).Should(Exit(0))
+}
+
+func enableServiceAccessForOrg(broker helpers.ServiceBroker, orgName string) {
+	Eventually(helpers.CF("enable-service-access", getServiceName(broker), "-o", orgName)).Should(Exit(0))
+}
+
+func getServiceName(broker helpers.ServiceBroker) string {
```

Comment:
Well, that indirection is not quite hard to follow. It's more or less self explanatory from the func name and it's signature what it's supposed to do.

We don't mind either way, if you insist and it's a common style around the CLI codebase we will modified it, but otherwise we don't see much value of doing this.

CC @nmaslarski

Reply:
Sounds good if its add to code readability.